mayastor/src/subsys/config/pool.rs

```diff
@@ -0,0 +1,181 @@
+            )
+        })?;
+
+        fs::write(&file, config.as_bytes())
+    }
+
+    /// Export current pool configuration
+    pub async fn export() -> Result<(), std::io::Error> {
+        static MUTEX: Lazy<Mutex<u32>> = Lazy::new(|| Mutex::new(0));
+
+        if let Some(file) = get_config_location() {
+            info!("Saving pool configuration to {}", file);
+            let _guard = MUTEX.lock().await;
+            return PoolConfig::get().write(file);
+        }
+
+        Ok(())
+    }
+
+    /// Create all pools specified in this pool configuration
+    async fn create_pools(&self) -> usize {
+        let mut failures = 0;
+        if let Some(pools) = self.pools.as_ref() {
+            for pool in pools {
+                info!("creating pool {}", pool.name);
+                if let Err(error) = Lvs::create_or_import(pool.into()).await {
+                    error!(
+                        "Failed to create pool {}: {}",
+                        pool.name,
+                        error.verbose()
+                    );
+                    failures += 1;
+                }
+            }
+        }
+        failures
+    }
+
+    /// Import pools
+    pub fn import_pools(self) {
+        assert_eq!(Cores::current(), Cores::first());
+        Reactor::block_on(async move {
+            let errors = self.create_pools().await;
+            if errors != 0 {
+                warn!(
+                    "Not all pools were imported successfully ({} errors)",
+                    errors
+                );
+            }
+        });
+    }
+
+    /// Capture current pool configuration
+    pub fn get() -> PoolConfig {
+        let pools = PoolsIter::new()
+            .map(|p| {
+                let base = p.get_base_bdev();
+                let name = p.get_name();
+                let replicas = ReplicaIter::new()
+                    .filter(|r| r.get_pool_name() == name)
+                    .map(|r| Replica {
+                        name: r.get_name().to_string(),
+                        share: r.get_share_type(),
+                    })
+                    .collect();
+                Pool {
+                    name: name.to_string(),
+                    disks: vec![base.bdev_uri().unwrap_or_else(|| base.name())],
+                    replicas: as_option(replicas),
+                }
+            })
+            .collect();
+
+        PoolConfig {
+            pools: Some(pools),
+        }
+    }
+}
+
+#[derive(Debug, Default, PartialEq, Serialize, Deserialize, Clone)]
+/// Pools that we create. Future work will include the ability to create RAID0
+/// or RAID5.
+pub struct Pool {
+    /// name of the pool to be created or imported
+    pub name: String,
+    /// bdevs to create outside of the nexus control
+    pub disks: Vec<String>,
+    /// list of replicas (not required, informational only)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub replicas: Option<Vec<Replica>>,
```

Comment:
I don't think we need the replicas, they're shared via the properties?

Reply:
As the comment says, the list of replicas is informational only - it is not used on start up as the pool knows what replicas have been created. This is potentially misleading, so for now the pool config is updated everytime a replica changes in order to maintain consistency. I would like to remove all references to replicas entirely, but I really want to do that as a separate PR in order to make it easier to reinstate should we wish to at some point in the future.

Reply:
:+1:

Reply:
Remove what is not needed, please. Just the pool information is what we want.

Reply:
No replica ingormation is generated. However if replica information is present in the yaml, rather than fail with invalid config, the replica information will be parsed but otherwise ignored.